src/ucp/proto/rndv.c

```diff
@@ -616,7 +616,8 @@ static void ucp_rndv_req_init_get_zcopy_lane_map(ucp_request_t *rndv_req)
 
             if ((lane_bw/max_lane_bw) <
                 (1. / context->config.ext.multi_lane_max_ratio)) {
-                lane_map &= ~UCS_BIT(lane_idx);
+                lane_map                                    &= ~UCS_BIT(lane_idx);
+                rndv_req->send.rndv_get.rkey_index[lane_idx] = UCP_NULL_RESOURCE;
```

Comment:
i think it still doesn't fix: lane_idx is relative to lane_map
need to fill rndv_req->send.rndv_get.rkey_index array only after we have the final lane_map

Reply:
@yosefe . not sure if I'm missing something. rkey_index is of size UCP_MAX_LANES and entries are added into ndv_req->send.rndv_get.rkey_index array (line:601) at an index where the corresponding lane_idx added to the lane_map.  send.rndv_get.rkey_index array can have holes and entry is unused if lane is removed from lane_map.

Reply:
ok